src/pages/scientificServices/imputation/views/JobHistory.tsx

```diff
@@ -1,20 +1,265 @@
         </div>
-      </div>
+        <div style={{ flex: 1, marginTop: '1rem' }}>
+          {pipelineRunsResponse ? (
+            <AutoSizer>
+              {({ width, height }) => (
+                // Sorting is unsupported on this table for now. Eventually
+                // we may update the paginated Teaspoons getAllPipelineRuns endpoint
+                // to support filters and sorting. Until then, the results will be
+                // sorted by creation date, with the most recent displayed first.
+                <FlexTable
+                  aria-label='job history table'
+                  width={width}
+                  height={height}
+                  rowHeight={55}
+                  rowCount={pipelineRunsResponse.results.length}
+                  columns={getColumns(pipelineRunsResponse.results)}
+                  noContentMessage={pipelineRunsResponse.totalResults > 0 ? ' ' : 'Nothing to display'}
+                  tabIndex={-1}
+                  variant={undefined}
+                />
+              )}
+            </AutoSizer>
+          ) : (
+            <Spinner />
+          )}
+        </div>
+        {!_.isEmpty(pipelineRunsResponse?.results) && (
+          <div style={{ marginBottom: '0.5rem' }}>
+            {/* @ts-ignore */}
+            <Paginator
+              filteredDataLength={pipelineRunsResponse?.totalResults ?? 0}
+              unfilteredDataLength={pipelineRunsResponse?.totalResults ?? 0}
+              pageNumber={pageNumber}
+              setPageNumber={(v) => {
+                setPageNumber(v);
+              }}
+              itemsPerPage={itemsPerPage}
+              setItemsPerPage={(v) => {
+                setPageNumber(1);
+                setItemsPerPage(v);
+              }}
+            />
+          </div>
+        )}
+      </main>
     </FooterWrapper>
   );
 };
+
+const getColumns = (paginatedRuns: PipelineRun[]) => {
+  return [
+    {
+      field: 'id',
+      headerRenderer: () => <HeaderCell>Job ID</HeaderCell>,
+      cellRenderer: ({ rowIndex }) => {
+        return <JobIdCell pipelineRun={paginatedRuns[rowIndex]} />;
+      },
+      size: { basis: 140 },
+    },
+    {
+      field: 'description',
+      headerRenderer: () => <HeaderCell>Description</HeaderCell>,
+      cellRenderer: ({ rowIndex }) => {
+        return <DescriptionCell pipelineRun={paginatedRuns[rowIndex]} />;
+      },
+      size: { basis: 140 },
+    },
+    {
+      field: 'status',
+      headerRenderer: () => <HeaderCell>Status</HeaderCell>,
+      cellRenderer: ({ rowIndex }) => {
+        return <StatusCell pipelineRun={paginatedRuns[rowIndex]} />;
+      },
+      size: { basis: 50 },
+    },
+    {
+      field: 'submitted',
+      headerRenderer: () => <HeaderCell>Submitted</HeaderCell>,
+      cellRenderer: ({ rowIndex }) => {
+        return <SubmittedCell pipelineRun={paginatedRuns[rowIndex]} />;
+      },
+      size: { basis: 90 },
+    },
+    {
+      field: 'completed',
```

Comment:
The [mock](https://www.figma.com/design/xMS9OAHL39AlqDjCI7O4rQ/Imputation-UX-mocks?node-id=597-4411&m=dev) has an additional column, "Deletion Date", which clarifies specifically when 2 weeks from the date of completion is.  Do when want that implemented here now, or later, or perhaps not at all?

Reply:
Good question- let's revisit this one a bit later. I have some questions about it but don't necessarily want to decide anything before merging this PR in